app/controllers/moods_controller.rb

```diff
@@ -24,9 +26,21 @@ def index
   # GET /moods/1
   # GET /moods/1.json
   def show
+    setup_stories
     redirect_to_path(moods_path) if @mood.user_id != current_user.id
   end
 
+  def tagged_moments_data
+    setup_stories
+    respond_to do |format|
+      if @moments
+        format.json do
+          render json: tagged_moments_data_json
```

Comment:
one-liner can work here as well. It might impact the specs, but it is neat
```suggestion
          render json: tagged_moments_data_json if @moments
```